core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -108,35 +119,72 @@ public void close() throws IOException {
      * the {@link #commit()} is called, to simplify coding.
      */
     public void abort() throws IOException {
-        close();
-        tmpFile.delete();
+        closeAndDeleteTempFile();
     }
 
     public void commit() throws IOException {
         close();
-        if (destFile.exists()) {
+        try {
+            // Try to make an atomic move.
+            Files.move(tmpPath, destPath, StandardCopyOption.ATOMIC_MOVE);
+        } catch (IOException e) {
```

Comment:
Possibly would be better to add a specific catch for `AtomicMoveNotSupported` just to have a clearer message (and in this case it is almost expected so we would not need to log it.  Still need to do exactly the same code afterwards though.

Reply:
>  Still need to do exactly the same code afterwards though.

yeah, that's why I didn't do that. But can add a private method for that purpose. Though my `if(destPath exists)` I put for that purpose: clarifying what could have gone wrong, but yeah.

Reply:
or just `if (e instanceof AtomicMoveNotSupported` { /* some logging */} else { /* some other logging */}` either works.

Reply:
Done